components/org.apache.stratos.cloud.controller/src/main/java/org/apache/stratos/cloud/controller/messaging/topology/TopologyBuilder.java

```diff
@@ -65,11 +67,31 @@ public static void handleServiceCreated(List<Cartridge> cartridgeList) {
                 if (!topology.serviceExists(cartridge.getType())) {
                     ServiceType serviceType = cartridge.isMultiTenant() ? ServiceType.MultiTenant : ServiceType.SingleTenant;
                     service = new Service(cartridge.getType(), serviceType);
-                    List<PortMapping> portMappings = cartridge.getPortMappings();
+                    List<PortMapping> portMappings = Arrays.asList(cartridge.getPortMappings());
                     Properties properties = new Properties();
-                    for (Map.Entry<String, String> entry : cartridge.getProperties().entrySet()) {
-                        properties.setProperty(entry.getKey(), entry.getValue());
+
+                    try {
+                        Property[] propertyArray = null;
+
+                        if (cartridge.getProperties() != null) {
+                            if (cartridge.getProperties().getProperties() != null){
+                                propertyArray = cartridge.getProperties().getProperties();
+                            }
+                        }
+
+                        List<Property> propertyList = new ArrayList<Property>();
```

Comment:
I would like to propose to move this logic to a method in Properties class so that it could be reused in other places as well. For the moment I'm merging this pull request, may be we can use a new pull request to do this improvement.
Thanks